include/TSPSolver.h

```diff
@@ -0,0 +1,20 @@
+#include "CostMatrix.h"
+
+struct TSPResult
+{
+    int cost;
+    std::vector<int> path;
+    TSPResult() = default;
```

Comment:
I dont think you need constructors for this struct. It simply holds two variables.
Intialize it this way: ```TSPResult result{cost, path};```